test/e2e/test/elasticsearch/checks_restart.go

```diff
@@ -58,7 +58,8 @@ func newNodeShutdownWatcher(es esv1.Elasticsearch) test.Watcher {
 			require.Empty(t, observedErrors)
 		},
 		func() bool {
-			return version.MustParse(es.Spec.Version).LT(shutdown.MinVersion)
+			// do not run this check on versions that do not support node shutdown or non-HA clusters where we restart all nodes at once
+			return version.MustParse(es.Spec.Version).LT(shutdown.MinVersion) && es.Spec.NodeCount() > 2
```

Comment:
Isn't it rather `|| es.Spec.NodeCount() < 3`?

We do a slightly more precise test for upgrades:
https://github.com/elastic/cloud-on-k8s/blob/eecd096008e7547526acd655da0ec48af3778899/test/e2e/test/elasticsearch/steps_mutation.go#L171-L177

```suggestion
			isNonHA := MustNumMasterNodes(es) < 3 || MustNumDataNodes(es) == 1
			return version.MustParse(es.Spec.Version).LT(shutdown.MinVersion) || isNonHA
```

Reply:
You are absolutely right! Monday morning brain not fully booted yet I guess.

Reply:
Would it make sense to introduce a new function and reuse it in both places?

```golang
// IsNonHA returns true if the provided Elasticsearch resource is not highly available by design. Either because it runs
// less than 3 master nodes or because there is only one data node.
func IsNonHA(es esv1.Elasticsearch) bool {
	return MustNumMasterNodes(es) < 3 || MustNumDataNodes(es) == 1
}
```

Reply:
Absolutely I will do that.